### ISSUE TITLE
No multi-component transform for JPEG2000 lossless

### ISSUE THREAD

**Issue by erikogabrielsson** [cgohlke/imagecodecs#88]
Hi,

From my understanding, multi-component transform (MCT) must be set for the openjpeg encoder to do RGB to YBR conversion, which produces better compression. If the encoding is lossy, the irreversible color transform (ICT) is used, and if lossless the reversible color transform (RCT). 

With imagecodecs, setting `mct=True` does not seem to produce YBR-converted compressed data when used together with lossless settings. I used the `Parse J2K codestream` code-snippet from [this issue](https://github.com/python-pillow/Pillow/issues/5485#issue-886396904) to parse out the MCT and DWT:

```python
jpeg2k_encoded_lossless = jpeg2k_encode(data, level=0, reversible=True, mct=True,  codecformat=JPEG2K.CODEC.J2K)
parse_jpeg2k(jpeg2k_encoded_lossless)
```

```
MCT: 0
DWT: 1
```

When setting `level=80` and `reversible=False`, MCT is applied:
```python
jpeg2k_encoded_lossy = jpeg2k_encode(data, level=80, reversible=False, mct=True, codecformat=JPEG2K.CODEC.J2K)
parse_jpeg2k(jpeg2k_encoded_lossy)
```
```
MCT: 1
DWT: 0
```

Comparing with Pillow (lossless):
```python
with io.BytesIO() as buffer:
    image.save(buffer, format='JPEG2000', irreversible=False, mct=1, no_jp2=True)
    pillow_encoded_lossless = buffer.getvalue()
parse_jpeg2k(pillow_encoded_lossless)
```
```
MCT: 1
DWT: 1
```

and lossy:
```python
with io.BytesIO() as buffer:
    image.save(buffer, format='JPEG2000', irreversible=True, mct=1, no_jp2=True)
    pillow_encoded_lossy = buffer.getvalue()
parse_jpeg2k(pillow_encoded_lossy)
```
```
MCT: 1
DWT: 0
```

I.e. Pillow will apply MCT for both lossless and lossy compression when `mct=1` is specified. 

Looking at the code, [MCT is not set when quality != 0](https://github.com/cgohlke/imagecodecs/blob/master/imagecodecs/_jpeg2k.pyx#L319). Is there a reason for this? Am I missing something?

**Comment by cgohlke** [comment]
I can't remember if there was a reason to disable MCT for lossless. Anyway, tests are passing here with MCT so it'll be enabled in the next version.

I'll also test if MCT can be enabled for > 3 components...

**Comment by cgohlke** [comment]
> multi-component transform (MCT) must be set for the openjpeg encoder to do RGB to YBR conversion, which produces better compression.

It's not that easy. In my limited testing, lossless compression without MCT sometimes produced smaller images compared to with MCT. One synthetic image was a factor 3.2 smaller...

**Comment by cgohlke** [comment]
Fixed in the v2024.1.1 release.
